.github/workflows/build-test-reusable.yml

```
@@ -356,7 +356,7 @@ jobs:
       - name: Run E2E test
         if: matrix.suite == 'rest'
         run: |
-          ${{ env.TRITON_TEST_CMD }} --inductor
+          timeout -s KILL 60m ${{ env.TRITON_TEST_CMD }} --inductor || ${{ inputs.ignore_errors }}
```

Note there is `--timeout=4800` (https://github.com/intel/intel-xpu-backend-for-triton/blob/main/scripts/test-triton.sh#L477). May be you want to adjust that timeout first?

---

> Note there is `--timeout=4800`

This timeout is in seconds.

---

```suggestion
          timeout -s KILL 4900 ${{ env.TRITON_TEST_CMD }} --inductor || ${{ inputs.ignore_errors }}
```

Even with timeout=300, it hangs until cancelled in 12 hours, but dies if SIGKILL sent externally. Maybe it is reasonable to set SIGKILL timer for `timeout+100` to allow either internal timeout with teardown, or external SIGKILL as a fallback.

---

There is still unanswered question about `success() || ${{ inputs.ignore_errors }}`

---

Yes, I am rechecking something :)